src/dbcp/helpers.py

```diff
@@ -90,12 +118,16 @@ def enforce_dtypes(df: pd.DataFrame, table_name: str, schema: str):
     except KeyError:
         raise KeyError(f"{table_name} does not exist in metadata.")
 
-    dtypes = {
-        col.name: SA_TO_PD_TYPES[str(col.type)]
-        for col in table.columns
-        if col.name in df.columns
-    }
-    return df.astype(dtypes)
+    for col in table.columns:
+        # Add the column if it doesn't exist
+        if col.name not in df.columns:
+            df[col.name] = None
+        df[col.name] = df[col.name].astype(SA_TO_PD_TYPES[str(col.type)])
```

Comment:
Pyarrow throws an error if a column exists in the schema but not in the dataframe.